pkg/apis/fleet/v1alpha1/types.go

```diff
@@ -64,6 +66,154 @@ type FleetSpec struct {
 
 	// TODO: support cluster selector?
 	// TODO: add options to allow customize fleet control plane if neccessary. And in future this could not be karmada.
+
+	// Plugin defines the plugins that would be installed in the fleet.
+	// +optional
+	Plugin *PluginConfig `json:"plugin,omitempty"`
+}
+
+type PluginConfig struct {
+	// Metric defines the configuration for the monitoring system installation and metrics collection..
+	// +optional
+	Metric *MetricConfig `json:"metric,omitempty"`
+	// Grafana defines the configuration for the grafana installation and observation.
+	// +optional
+	Grafana *GrafanaConfig `json:"grafana,omitempty"`
+}
+
+type MetricConfig struct {
+	// Thanos defines the configuration for the thanos querier and store gateway.
+	Thanos ThanosConfig `json:"thanos,omitempty"`
+	// Prometheus defines the configuration for the prometheus installation
+	// in the clusters observed by the thanos,
+	// by default thanos sidecar will be installed in thanos sidecar mode.
+	Prometheus PrometheusConfig `json:"prometheus,omitempty"`
+}
+
+type PrometheusConfig struct {
+	// Chart defines the helm chart config of the prometheus.
+	// default values is
+	//
+	// chart:
+	//   repository: oci://registry-1.docker.io/bitnamicharts
+	//   name: kube-prometheus
+	//   version: 8.9.1
+	//
+	// +optional
+	Chart *ChartConfig `json:"chart,omitempty"`
+	// NodeExporter defines the configuration for the node exporter.
+	// +optional
+	NodeExporter *PrometheusExporterConfig `json:"nodeExporter,omitempty"`
+	// KubeStateMetrics defines the configuration for the kube-state-metrics.
+	// +optional
+	KubeStateMetrics *PrometheusExporterConfig `json:"kubeStateMetrics,omitempty"`
+	// ExtraArgs is the set of extra arguments for Prometheus chart.
+	//
+	// For Example, using following configuration to create a ServiceMonitor to monitor prometheus itself.
+	// extraArgs:
+	//   prometheus:
+	//     serviceMonitor:
+	//       enabled: true
+	//
+	// +optional
+	ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
+}
+
+type PrometheusExporterConfig struct {
+	// Enabled indicates whether the exporters are enabled.
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+type ChartConfig struct {
+	// Repository defines the repository of chart.
+	// OCI registry is not supported.
```

Comment:
Why below comments for both grafana and thanos is `OCI:`

Reply:
nit, removed